addons/website_theme_install/models/theme_models.py

```diff
@@ -161,10 +161,22 @@ def _post_copy(self, mod, website=False):
     @api.model
     def _toggle_view(self, xml_id, active):
         obj = self.env.ref(xml_id)
+        website = self.env['website'].get_current_website()
         if obj._name == 'theme.ir.ui.view':
-            website = self.env['website'].get_current_website()
             obj = obj.with_context(active_test=False)
             obj = obj.copy_ids.filtered(lambda x: x.website_id == website)
+        else:
+            # If a theme post copy wants to enable/disable a view, this is to
+            # enable/disable a given functionality which is disabled/enabled
+            # by default. So if a post copy asks to enable/disable a view which
+            # is already enabled/disabled, we would not consider it otherwise it
+            # would COW the view for nothing.
+            has_specific = self.env['ir.ui.view'].search_count([
+                ('key', '=', obj.key),
```

Comment:
What if the view that has the given xml_id does not have a key?
has_specific might always be true then since there is a good chance there is at least a view without a key